### DISCUSSION TITLE
Why don't you add ML.NET examples along with Python and R?

### DISCUSSION THREAD

**Discussion by 64J0** [microsoft/ML-For-Beginners#693] [Ideas]
Since this is a project from Microsoft, why we don't have examples using [ML.NET](https://dotnet.microsoft.com/en-us/apps/machinelearning-ai/ml-dotnet), which is their .NET package for ML?

**Comment by Shaikhasna** [comment]
Hi @64J0,
Thanks for raising this! ML.NET is indeed Microsoft’s .NET framework for machine learning, but most open-source beginner-focused ML tutorials prioritize Python and R because:

Community support – Python has a massive ecosystem (scikit-learn, PyTorch, TensorFlow) and rich beginner-friendly resources.

Cross-platform usability – Python and R notebooks run easily across Windows, Linux, and MacOS, whereas ML.NET is tightly coupled with .NET and Windows-centric environments.

Integration in curriculum – ML-For-Beginners targets learners with no prior coding experience, and Python/R are easier for interactive notebooks.
That said, adding ML.NET examples could be a valuable enhancement for .NET developers. A possible contribution could include a small ML.NET translation of one regression/classification notebook, which would complement the existing Python and R examples.

I’d be happy to experiment with a simple ML.NET example and submit a PR if the maintainers find it useful!

**Reply by 64J0** [reply]
Got it, thanks for the explanation!
